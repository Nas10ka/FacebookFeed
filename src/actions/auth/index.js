import types from './types';

export default Object.freeze({
    login: (user) => ({
        type:    types.LOGIN,
        payload: user,
    }),
    loginSucceed: () => ({
        type: types.LOGIN_SUCCEED,
    }),
    loginFailed: (error) => ({
        type:    types.LOGIN_FAILED,
        payload: error,
        error:   true,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
    logoutSucceed: () => ({
        type: types.LOGOUT_SUCCEED,
    }),
    logoutFailed: (error) => ({
        type:    types.LOGOUT_FAILED,
        payload: error,
        error:   true,
    }),
    authenticate: (token) => ({
        type:    types.AUTHENTICATE,
        payload: token,
    }),
    authenticateSucceed: () => ({
        type: types.AUTHENTICATE_SUCCEED,
    }),
    authenticateFailed: (error) => ({
        type:    types.AUTHENTICATE_FAILED,
        payload: error,
        error:   true,
    }),
    signup: (signUp) => ({
        type:    types.SIGNUP,
        payload: {
            firstName: signUp.firstName,
            lastName:  signUp.lastName,
            password:  signUp.password,
            email:     signUp.email,
            invite:    signUp.invite,
        },
    }),
    signupSucceed: () => ({
        type: types.SIGNUP_SUCCEED,
    }),
    signupFailed: (error) => ({
        type:    types.SIGNUP_FAILED,
        payload: error,
        error:   true,
    }),
});
