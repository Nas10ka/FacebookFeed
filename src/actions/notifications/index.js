import types from './types';
import { v4 } from 'uuid';

export default Object.freeze({
    invoke: (error) => ({
        type:    types.INVOKE,
        payload: {
            id: v4(),
            error,
        },
    }),
    dissolve: (id) => ({
        type:    types.DISSOLVE,
        payload: id,
    }),
});
