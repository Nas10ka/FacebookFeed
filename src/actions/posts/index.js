//Instructions
import types from './types';
import { api, groupId } from '../../instruments/api';
import { normalize } from 'normalizr';
import { post } from '../../shemas';
import usersActions from '../users';
export default Object.freeze({
    fetchPosts: () => async (dispatch) => {
        try {
            const response = await fetch(`${api}/feed?size=30`, {
                method:  'GET',
                headers: {
                    'x-no-auth': groupId,
                },
            });

            const { data, message } = await response.json();

            console.log('data', data);
            if (response.status !== 200) {
                throw new Error(message);
            }

            const normilizedPosts = normalize(data, [post]);

            dispatch(usersActions.fillUsers(normilizedPosts.entities.users));

            dispatch({
                type:    types.FETCH_POSTS_SUCCEED,
                payload: normilizedPosts,
            });
        } catch (error) {
            dispatch({
                type:    types.FETCH_POSTS_FAILED,
                payload: error.message,
                error:   true,
                meta:    'Additional info',
            });
        } finally {
            console.log('finally');
        }
    },
    clearPosts: () => ({
        type: types.CLEAR_POSTS,
    }),

    createPost: (comment) => ({
        type:    types.CREATE_POST,
        payload: comment,
    }),
    createPostSucceed: (newPost) => ({
        type:    types.CREATE_POST_SUCCEED,
        payload: newPost,
    }),
    createPostFailed: (error) => ({
        type:    types.CREATE_POST_FAILED,
        payload: error,
        error:   true,
    }),

    deletePost: (id) => ({
        type:    types.DELETE_POST,
        payload: id,
    }),
    deletePostSucceed: (id) => ({
        type:    types.DELETE_POST_SUCCEED,
        payload: id,
    }),
    deletePostFailed: (error) => ({
        type:    types.DELETE_POST_FAILED,
        payload: error,
        error:   true,
    }),

    likePost: (id) => ({
        type:    types.LIKE_POSTS,
        payload: id,
    }),
    likePostSucceed: (id) => ({
        type:    types.LIKE_POSTS_SUCCEED,
        payload: id,
    }),
    likePostFailed: (error) => ({
        type:    types.LIKE_POSTS_FAILED,
        payload: error,
        error:   true,
    }),
    dislikePost: (id) => ({
        type:    types.DISLIKE_POSTS,
        payload: id,
    }),
    dislikePostSucceed: (id) => ({
        type:    types.DISLIKE_POSTS_SUCCEED,
        payload: id,
    }),
    dislikePostFailed: (error) => ({
        type:    types.DISLIKE_POSTS_FAILED,
        payload: error,
        error:   true,
    }),
});
