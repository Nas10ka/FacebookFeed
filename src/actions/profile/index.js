import types from './types';

export default Object.freeze({
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
    updateProfile: (user) => ({
        type:    types.UPDATE_PROFILE,
        payload: user,
    }),
    updateProfileSucceed: (user) => ({
        type:    types.UPDATE_PROFILE_SUCCEED,
        payload: user,
    }),
    updateProfileFailed: (error) => ({
        type:    types.UPDATE_PROFILE_FAILED,
        payload: error,
        error:   true,
    }),
    updateAvatar: (avatar) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatar,
    }),
    updateAvatarSucceed: (url) => ({
        type:    types.UPDATE_AVATAR_SUCCEED,
        payload: url,
    }),
    updateAvatarFailed: (error) => ({
        type:    types.UPDATE_AVATAR_FAILED,
        payload: error,
        error:   true,
    }),
});
