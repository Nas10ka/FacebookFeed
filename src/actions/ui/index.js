//Core
import types from './types';

export default Object.freeze({
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    startAuthFetching: () => ({
        type: types.START_AUTH_FETCHING,
    }),
    stopAuthFetching: () => ({
        type: types.STOP_AUTH_FETCHING,
    }),
    startFeedFetching: () => ({
        type: types.START_FEED_FETCHING,
    }),
    stopFeedFetching: () => ({
        type: types.STOP_FEED_FETCHING,
    }),
    startProfileFetching: () => ({
        type: types.START_PROFILE_FETCHING,
    }),
    stopProfileFetching: () => ({
        type: types.STOP_PROFILE_FETCHING,
    }),
    startAvatarFetching: () => ({
        type: types.START_AVATAR_FETCHING,
    }),
    stopAvatarFetching: () => ({
        type: types.STOP_AVATAR_FETCHING,
    }),
    startProfileEditing: () => ({
        type: types.START_PROFILE_EDITING,
    }),
    stopProfileEditing: () => ({
        type: types.STOP_PROFILE_EDITING,
    }),
});
