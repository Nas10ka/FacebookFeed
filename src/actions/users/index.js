import types from './types';

export default Object.freeze({
    fillUsers: (users) => ({
        type:    types.FILL_USERS,
        payload: users,
    }),
});
