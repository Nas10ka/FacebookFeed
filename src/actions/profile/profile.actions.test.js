//Core
import actions from './';

const user = {
    firstName: 'John',
    lastName:  'Snow',
};

const avatar = 'avatar';
const error = 'error';
const url = 'url';

describe('Profile actions:', () => {
    test('Fill Profile action', () => {
        expect(actions.fillProfile(user)).toMatchSnapshot();
    });
    test('Update Profile action', () => {
        expect(actions.updateProfile(user)).toMatchSnapshot();
    });
    test('Update Profile Succeed action', () => {
        expect(actions.updateProfileSucceed(user)).toMatchSnapshot();
    });
    test('Update Profile Failed action', () => {
        expect(actions.updateProfileFailed(error)).toMatchSnapshot();
    });
    test('Update Profile action', () => {
        expect(actions.updateAvatar(avatar)).toMatchSnapshot();
    });
    test('Update Avatar Succeed action', () => {
        expect(actions.updateAvatarSucceed(url)).toMatchSnapshot();
    });
    test('Update Avatar Failed action', () => {
        expect(actions.updateAvatarFailed(error)).toMatchSnapshot();
    });
    test('Clear Profile action', () => {
        expect(actions.clearProfile()).toMatchSnapshot();
    });
});
