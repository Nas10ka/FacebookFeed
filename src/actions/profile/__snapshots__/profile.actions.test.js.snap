// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile actions: Clear Profile action 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`Profile actions: Fill Profile action 1`] = `
Object {
  "payload": Object {
    "firstName": "John",
    "lastName": "Snow",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`Profile actions: Update Avatar Failed action 1`] = `
Object {
  "error": true,
  "payload": "error",
  "type": "UPDATE_AVATAR_FAILED",
}
`;

exports[`Profile actions: Update Avatar Succeed action 1`] = `
Object {
  "payload": "url",
  "type": "UPDATE_AVATAR_SUCCEED",
}
`;

exports[`Profile actions: Update Profile Failed action 1`] = `
Object {
  "error": true,
  "payload": "error",
  "type": "UPDATE_PROFILE_FAILED",
}
`;

exports[`Profile actions: Update Profile Succeed action 1`] = `
Object {
  "payload": Object {
    "firstName": "John",
    "lastName": "Snow",
  },
  "type": "UPDATE_PROFILE_SUCCEED",
}
`;

exports[`Profile actions: Update Profile action 1`] = `
Object {
  "payload": Object {
    "firstName": "John",
    "lastName": "Snow",
  },
  "type": "UPDATE_PROFILE",
}
`;

exports[`Profile actions: Update Profile action 2`] = `
Object {
  "payload": "avatar",
  "type": "UPDATE_AVATAR",
}
`;
